diff --git a/src/datatrove/utils/text.py b/src/datatrove/utils/text.py
index 7ab7d3d..233940b 100644
--- a/src/datatrove/utils/text.py
+++ b/src/datatrove/utils/text.py
@@ -259,6 +259,8 @@ def simplify_text(text: str, config=DEF_TEXT_NORM_CONFIG) -> str:
 
 # from https://tedboy.github.io/nlps/_modules/nltk/util.html#ngrams
 def ngrams(sequence: Iterable, n: int):
+    if isinstance(sequence, list) and len(sequence) < n:
+        sequence += [""] * (n - len(sequence))
     iterables = tee(sequence, n)
 
     for i, sub_iterable in enumerate(iterables):  # For each window,
